packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -268,21 +309,6 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
 		// If GC is disabled, don't set wip used routes.
 		if (!this.gcDisabled) {
 			wipSerializedUsedRoutes = this.wipSerializedUsedRoutes;
-			/**
-			 * The absence of wip used routes indicates that GC was not run on this node. This can happen if:
-			 * 1. A child node was created after GC was already run on the parent. For example, a data store
-			 * is realized (loaded) after GC was run on it creating summarizer nodes for its DDSes. In this
-			 * case, the used routes of the parent should be passed on the child nodes and it should be fine.
-			 * 2. A new node was created but GC was never run on it. This can mean that the GC data generated
-			 * during summarize is complete . We should not continue, log and throw an error. This will help us
-			 * identify these cases and take appropriate action.
-			 */
-			if (wipSerializedUsedRoutes === undefined) {
```

Comment:
Now that this if check is gone, the block below can have `wipSerializedUsedRoutes` be `undefined`.  But we do a non-null assertion and pass it to `new SummaryNodeWithGC`... We should tidy this up somehow.

Reply:
Probably just change the type of the constructor param / property to be `string | undefined`.  We already have `undefined` checks throughout that code.

Reply:
Good point. The undefined checks were added over time due to bugs so it makes sense for the type to be string | undefined. Changed it.